packages/connect/src/ScProvider.ts

```diff
@@ -95,7 +95,10 @@ export class ScProvider implements ProviderInterface {
       isExtension,
       chainSpec,
       parachainSpec,
-    ).then((provider) => (this.#provider = provider))
+    ).then(async (provider) => {
+      await provider.connect()
+      return (this.#provider = provider)
+    })
```

Comment:
I would suggest doing this, instead, which better aligns with the behavior of the `WsProvider`:

```ts
  public constructor(
    displayName: string,
    chainSpec: string,
    parachainSpec?: string,
    autoConnect?: boolean = true,
  ) {
    const isExtension = !!document.getElementById("substrateExtension")
    this.#providerP = this.internalProvider(
      displayName,
      isExtension,
      chainSpec,
      parachainSpec,
    )
    
    if (autoConnect) this.#providerP.then(provider => provider.connect())
  }
```